src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -535,24 +538,20 @@ private function compileSwitchDefault(bool $hasVars, bool $matchHost): string
         } else {
             $code = '';
         }
-        if ($this->supportsRedirections) {
-            $code .= <<<EOF
 
-            if (\$requiredSchemes && !isset(\$requiredSchemes[\$context->getScheme()])) {
-                if ('GET' !== \$canonicalMethod) {
-                    \$allow['GET'] = 'GET';
-                    break;
-                }
+        $code .= <<<EOF
 
-                return \$this->redirect(\$rawPathinfo, \$ret['_route'], key(\$requiredSchemes)) + \$ret;
+            if (\$requiredMethods && !isset(\$requiredMethods[\$canonicalMethod]) && !isset(\$requiredMethods[\$requestMethod])) {
+                \$allow += \$requiredMethods;
```

Comment:
By checking requirements before methods, we fixed an issue where a 405 is triggered while a scheme requirement is not met. Shouldn't we keep this fixed behavior, ie check methods last?

Reply:
I also thought about this alot. There are cases were it's not optimal in both cases.

Request GET http /
Route POST  https /

If you check the scheme first instead of the method. It will try to make a redirect from http to https although the route requires POST which is not met.

So I think checking method first is better as getting a 405 instead of a 404 for the case you explained does not really matter. Also this order is what was implemented by UrlMatcher. So it makes it more consistent.

Reply:
Here is why I was talking about that: report in #22739, fix just sent in #26312.